broadinstitute/pearl
# AR-87: participant consent flow

Participants can now see consent tasks in their task list, and click them to go to the consent form for completion.  Note that submitting the consent forms is NOT yet implemented.  

The data flow model this uses is that the hub/task list are loaded, which gives the client enough information to render the links to the consent form.  Once the participant travels to that link, an API endpoint is called to retrieve both the form configuration and any past responses by the participant.  This will be the same approach we use for surveys.  This should be more efficient and robust than the previous approach of loading all participant responses and all study form content immediately upon the participant entering the hub.

This also adds capability for populating synthetic participants with pre-populated tasks.  A new participant, `newbie@test.com` has been added to represent a participant who just finished enrolling in the study, and has a consent task queued up. 

TO TEST:
1. nuke/reset db
2. run populate_setup.sh and populate_portal.sh ourhealth
3. go to `sandbox.ourhealth.localhost:3001`
4. log in as `newbie@test.com`
5. confirm you see rudimentary task list
![image](https://user-images.githubusercontent.com/2800795/216689828-4502f4a2-3915-4a96-aef5-06bf8a3f51fd.png)
6. Click on the consent task
7. confirm you are taken to the consent form, and are able to page forward/backwards (submitting the form is not yet implemented
![image](https://user-images.githubusercontent.com/2800795/216690053-a1de1efd-422a-45d0-9707-6e168d5550b8.png)
